ovs: Regenerate iface metadata after RouteRule metadata generation

cathay4t opened the pull request:
When desire state only contain route rule for OVS interface route table
ID, NetworkManager will complains:

    A connection with a 'ovs-interface' setting must have a master.

The root cause is because interface metadata is not generated if desire
state has no interface at all.

The fix is regenerate interface metadata after route rule metadata
generation.

Integration test case added and marked as tier1 due to RHV requirement.

ffmancera reviewed (changes requested):
The patch is not fixing the issue, it is failing on the CI. Please, could you check it out?

cathay4t committed:
ovs: Regenerate iface metadata after RouteRule metadata generation

When desire state only contain route rule for OVS interface route table
ID, NetworkManager will complains:

    A connection with a 'ovs-interface' setting must have a master.

The root cause is because interface metadata is not generated if desire
state has no interface at all.

The fix is regenerate interface metadata on changed/desired interface
after route rule metadata generation.

Integration test case added and marked as tier1 due to RHV requirement.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
